lib/matplotlib/cbook.py

```diff
@@ -1634,6 +1634,8 @@ def _safe_first_finite(obj, *, skip_nonfinite=True):
     def safe_isfinite(val):
         if val is None:
             return False
+        if isinstance(val, int):
```

Comment:
I'm not sure special-casing ints here make sense -- certainly that'll make a microbenchmark based on plotting ints faster, but at the cost of introducing a branch for all other cases (plotting floats is likely much more common, and even np.int is perhaps a more common case than python ints in real code).

If one really wants to workaround the fact that np.isfinite is relatively slow, one can instead use math.isfinite (which on a quick microbenchmark is extremely fast), taking into account the fact that it won't handle certain cases like datetimes, so something like
```python
try:
    if math.isfinite(val): return True
except TypeError:
    pass
# continue with the np.isfinite check
```

Reply:
This is an excellent suggestion. On my system `np.isfinite(val) if np.isscalar(val) else True` is about 900 ns, and `math.isfinite(val)` (including the try-except) is 80 ns. The `isinstance(val, int)` is 50 ns, but does not handle the important float case.